frontend/simple/views/Join.vue

```diff
@@ -97,36 +97,42 @@
     </section>
 </template>
 <script>
+import {namespace} from '../js/backend/hapi'
 import * as Events from '../../../shared/events'
 import backend from '../js/backend/'
-import * as db from '../js/database'
-import {latestContractState} from '../js/state'
+import { latestContractState } from '../js/state'
 import L from '../js/translations'
+
 export default {
   name: 'Join',
-  mounted () {
-    this.fetchData()
+  async mounted () {
+    let state = await latestContractState(this.$route.query.groupId)
+    console.log(state)
+    if (!state.invitees.find(invitee => invitee === this.$store.state.loggedIn.name)) {
+      // TODO: proper user-facing error
+      console.log(new Error('Invalid Invitation'))
+      this.$router.push({path: '/mailbox'})
+    }
+    for (var i = 0; i < state.members.length; i++) {
+      const contractId = await namespace.lookup(state.members[i])
+      state.members[i] = await latestContractState(contractId)
```

Comment:
These are top-level awaits, meaning that a try/catch is required to handle any errors gracefully here.